### ISSUE TITLE
Awesome job!!! Let's address the iPhone X

### ISSUE THREAD

**Issue by thexande** [aunnnn/AppStoreiOS11InteractiveTransition#1]
Greetings, this is by far the best custom implementation of this feature I have seen. 

The non X devices look near perfect, but as you listed in the readme, there is a issue with the iPhone X in the top layout, as well as the end of the transition back to the Today view. 

I'm looking into it, and I was wondering if you have any ideas for places to start.

Thanks!!

![app store](https://user-images.githubusercontent.com/7704414/44552692-4bd9f980-a6e8-11e8-9c71-aa58b14d8b3d.gif)

**Comment by aunnnn** [comment]
Thanks for checking it out! Will look at it soon.

I ran this on iPhone X with iOS 12 and it seems to fix the issue, don't know why... (but the notch still obscure top portion of card content)

For iOS 11, a dirty fix is to tick `Safe Area Layout Guide` off in `CardContentView.xib` file (Both root view and the 'card container' view that actually holds card's content), to prevent any effects from safe area.

But the downside is that the content/secondary title on card will still be obscured by the notch. Although we can still retain safe area inset after presented by ticking `Safe Area Layout Guide` off only on the root xib (leaving safe area active on the card container view), this will cause _noticable abrupt layout change_ just before the presenting animation begins.

The middle ground is to smoothly animate top inset of the card (from little inset to final safe area insets), but will need some time for this one.

**Comment by salvatoreboemia** [comment]
Hi, I have a question. It's possible to add a "x" button for close? like AppStore? And for iOS 10 and 12 work? thanks

**Comment by prajwalsprakash** [comment]
> Hi, I have a question. It's possible to add a "x" button for close? like AppStore? And for iOS 10 and 12 work? thanks

Yeah, just add a floating button in detail xib don't make it part of card content view. It must be above that. and handle viewcontroller dismiss on tap of the button.
